CMakeLists.txt

```diff
@@ -2,13 +2,17 @@
 # General setup
 #
 cmake_minimum_required(VERSION 3.10)
-set(CMAKE_OSX_ARCHITECTURES "x86_64")
+
 # Minimum OS X version.
 # This is inserted into the Info.plist as well.
 
-# MacOS prior to 10.12 did not fully support C++17, which is used to
-# handle configuration options
-set(CMAKE_OSX_DEPLOYMENT_TARGET "10.12.0" CACHE STRING "")
+# MacOS prior to 10.14 did not support aligned alloc which is used to implement
+# std::unique_ptr in the arm64 STL. x86_64 builds can override this to 10.12.0
```

Comment:
Remembering that arm64 targets start at 11.0 anyway. So it may be worth figuring out a way to set this target per architecture.

Reply:
> Remembering that arm64 targets start at 11.0 anyway. So it may be worth figuring out a way to set this target per architecture.

I spent a few hours on this, but it doesn't seem to be possible due to some unfortunate limitations in CMake. 

Basically, the target architecture isn't know by the CMakeLists script until the project(..) call, but CMake ignores the settings of the CMAKE_OSX_DEPLOYMENT_TARGET after the project(..) call.